shtdlib.sh

```diff
@@ -357,18 +394,36 @@ function compare_versions {
     return $(( lowest_ver_line-1 ))
 }
 
-# Prints the version of a command, accepts 1-4 parameters
+# Set conveniece variable for bash v4 compat
+if compare_versions "${BASH_VERSION}" "4" ; then
+    bash_pre_v4=true
+else
+    bash_pre_v4=false
+fi
+
+# Set timeout value to use for read, v3 does not support decimal seconds
+if "${bash_pre_v4}" ; then
+    read_timeout='1'
+else
+    read_timeout='0.1'
+fi
+
+# Prints the version of a command, arguments include:
 # 1. Full or relative path to command (required)
 # 2. Text to display before version info (optional)
-# 3. Flag/Argument to command to get version (optional, defaults to --version)
-# 4. Error message if command is not found, to ignore redirect stderr like
-# this:     print_version bash 2> /dev/null
+# 3+. Flag(s)/Argument(s) to command to get version (optional, defaults to --version)
+# error_msg variable: Error message if command is not found, to ignore redirect
+# stderr run this like so:    print_version bash 2> /dev/null
 function print_version {
     local error_msg
     error_msg="${4:-Unable to find command ${1}}"
```

Comment:
You removed the 4th argument in the comments but still used here???

Reply:
`s/4/error_msg/`?

Reply:
thx, fixed